### PR TITLE
Refactoring - st72710 (Jan Stehno)

### PR DESCRIPTION
**Pull request by janstehno**
- created class for polynomial
  - changed `Poly` to `Polynomial`
  - renamed attributes, methods, and variables
- created class for complex number
  - changed `Cplx` to `ComplexNumber`
  - renamed attributes, methods, and variables
- created class for fractal generation
  - created `Fractal`
  - extracted and renamed readonly attributes
  - extracted and renamed other used attributes
  - split `Main` to smaller methods
- changed tests accordingly
- removed unused and/or commented code
- program should be executable

### CODE REVIEW

File: NNPTPZ1/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            double denominator = c.RealNumber * c.RealNumber + c.ImaginaryUnit * c.ImaginaryUnit;
+
+            return new ComplexNumber()
+            {
+                RealNumber = dividend.RealNumber / denominator,
+                ImaginaryUnit = dividend.ImaginaryUnit / denominator
+            };
+        }
+
+        public double GetAbsolute()
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: wrong name, absolute value is technical term in complex numbers and should be preserved without any abbreviations or shortenings

File: NNPTPZ1/Fractal.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+using System.Collections.Generic;
+using System.Drawing;
+
+using Mathematics;
+
+namespace App
+{
+    class Fractal
+    {
+        internal readonly int NUMBER_OF_ITERATIONS = 30;
```
Comment (rodi0878) [inline NNPTPZ1/Fractal.cs:11]: there is too much fields and in many cases you prefer this mess over direct return values from methods, this mades it really bad for comprehension of whats going on here...

File: NNPTPZ1/Fractal.cs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                }
+            }
+        }
+
+        internal void FindWorldCoordinationsOf(int i, int j)
+        {
+            double y = ymin + i * ystep;
+            double x = xmin + j * xstep;
+
+            complexNumber = new ComplexNumber()
```
Comment (rodi0878) [inline NNPTPZ1/Fractal.cs]: this should be return value

File: NNPTPZ1/Fractal.cs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            {
+                RealNumber = x,
+                ImaginaryUnit = y
+            };
+        }
+
+        internal void FindSolutionByNewtonIteration()
+        {
+            iterations = 0;
```
Comment (rodi0878) [inline NNPTPZ1/Fractal.cs]: this should be return value

File: NNPTPZ1/Fractal.cs [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                {
+                    knownRoot = true;
+                    position = i;
+                    break;
+                }
+            }
+            if (!knownRoot)
+            {
+                roots.Add(complexNumber);
+                position = roots.Count;
```
Comment (rodi0878) [inline NNPTPZ1/Fractal.cs:138]: this should be return value

**Review by rodi0878 — CHANGES_REQUESTED** [review]
please see attached comments and provide update. current version of code needs some additional tweaks

**Comment by janstehno** [comment]
- made requested changes

Just to understand this, it is better to use method parameters and return values rather than global variables?

I can understand it for `position`, `iterations` and `currentComplexNumber`. But for me it does not make sense to do it the same way for `polynomial`, `derivatedPolynomial`, or `bitmap` for example. It is possible, sure, but I would have to pass them as parameters too many times in my opinion.

**Comment by rodi0878** [comment]
generally you should not try to hide any input/output from functions. Parameters should be preferred instead of fields/properties for passing of directly related data.

ok accepted

**Pull request closed without merging**